### ISSUE TITLE
Possible bug in y limit calculation in stripplot/swarmplot with small values

### ISSUE THREAD

**Issue by mojones** [mwaskom/seaborn#1738]
Given a dataframe with one categorical variable and two continuous variables:

```python
import random
test = pd.DataFrame({
    'foo' : [random.choice('ab') for _ in range(100)],
    'bar' : [random.random()  for _ in range(100)],
    'baz' : [random.random() / 10000 for _ in range(100)]
})
```
drawing a stripplot for the continuous variable with a range of 0-1 we get sensible y axis limits:

```python
sns.stripplot(
    data = test,
    x = 'foo',
    y = 'bar'
)
```

![Selection_007](https://user-images.githubusercontent.com/893512/57083289-31810280-6cf0-11e9-875b-c8f035b33374.png)

but drawing the same plot with the variable that has a very small range of 0-0.0001:

```python
sns.stripplot(
    data = test,
    x = 'foo',
    y = 'baz'
)
```

The y limits are set much wider than the span of the values and so they are all compressed into a small area of the chart:

![Selection_008](https://user-images.githubusercontent.com/893512/57083774-fd5a1180-6cf0-11e9-81cb-567f0277970d.png)

Setting the y limits manually gives closer to the expected result:

```python
sns.stripplot(
    data = test,
    x = 'foo',
    y = 'baz'
)
plt.ylim(test['baz'].min(), test['baz'].max())
```

![Selection_009](https://user-images.githubusercontent.com/893512/57083921-398d7200-6cf1-11e9-88b2-ad0ed56c015c.png)

Is this expected behaviour? I get the same effect with `swarmplot`, although other categorical plots like `boxplot` give me the y limits that I expect:

```python
sns.boxplot(
    data = test,
    x = 'foo',
    y = 'baz'
)
```

![Selection_010](https://user-images.githubusercontent.com/893512/57084078-883b0c00-6cf1-11e9-9956-0101103d8a98.png)

**Comment by mwaskom** [comment]
It's expected in the sense that there is a longstanding matplotlib issue in the interaction between `scatter` and autoscaling: https://github.com/matplotlib/matplotlib/issues/6015

**Comment by mojones** [comment]
Thanks for the pointer, I'll add this to the long list of matplotlib idiosyncrasies to keep track of.

**Comment by mojones** [comment]
For anyone else who ends up here, this is fixed in matplotlib 3.2
